### DISCUSSION TITLE
resolve mx record using dns library of node in vercel-edge-function?

### DISCUSSION THREAD

**Discussion by bmbshlly** [vercel/community#860] [Help] [answered]
So I want to resolve MX record in edge for fast responses. The vercel-edge-function runtime for nextjs only supports node web APIs. I am using 
`import { promises } from "dns";`
promises.resolveMx function in node. Have a better solution for DNS resolution?

**Comment by amyegan** [comment, marked as answer]
Hi @bmbshlly! [Node.js APIs are not available](https://vercel.com/docs/concepts/functions/edge-functions/limitations#node.js-apis-are-not-available) in Edge Functions. [Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions#) might be a better option for your needs.

**Reply by bmbshlly** [reply]
Hi, @amyegan thanks for the quick reply. I don't want to fall short on edge latency, can you help with some other solution?

**Reply by amyegan** [reply]
I don't have another solution that would allow you to use a Node API in an Edge Function. But you can call a REST API to get data like 
```
import { NextResponse } from "next/server";

export default (req) => {
  return fetch("https://example.com/api/v2/data")
    .then((response) => response.json())
    .then((data) => {
      return NextResponse.json(data);
    });
};

export const config = {
  runtime: "experimental-edge",
};
```

If you can find, or build, an external service to get the data you want, then you could make it work with Edge Functions.

More information is available in the [documentation](https://vercel.com/docs/concepts/functions/edge-functions#key-takeaways):
>When using Edge Functions with Next.js (as API Routes configured to run at the edge), you can import and use the [next/server](https://vercel.com/docs/concepts/functions/edge-functions/quickstart#add-code-to-your-function) helpers (NextRequest, NextResponse)

>When using Edge Functions, you can add the [@vercel/edge package](https://vercel.com/docs/concepts/functions/edge-functions/edge-functions-api#the-@vercel/edge-package) and import helper functions such as **geolocation**, **rewrite**, and **ipAddress**
